lib/utils/extractMagicBlocks.ts

```diff
@@ -21,9 +21,12 @@ export function extractMagicBlocks(markdown: string) {
   let index = 0;
 
   const replaced = markdown.replace(MAGIC_BLOCK_REGEX, match => {
-    // Use backticks so it becomes a code span, preventing remarkParse from
-    // parsing special characters in the token as markdown syntax
-    const token = `\`__MAGIC_BLOCK_${index}__\``;
+    /**
+     * - Use backticks so it becomes a code span, preventing remarkParse from parsing 
+     *   special characters in the token as markdown syntax
+     * - Prepend a newline to the token to ensure it is parsed as a block level node
+     */
+    const token = `\n\`__MAGIC_BLOCK_${index}__\``;
```

Comment:
Looks like some snapshots are failing due to the change. Should we be conditionally adding this based on whether or not there is already a newline character, or should we be stripping it again afterwards? Or does it not actually matter?